src/compose/ComposeBox.js

```diff
@@ -69,6 +69,11 @@ type Props = $ReadOnly<{|
   defaultMessage?: string,
   defaultTopic?: string,
 
+  autoFocusTopic: boolean,
+  autoFocusMessage: boolean,
+  useTopicAutocomplete: boolean,
+  alwaysShowTopic: boolean,
```

Comment:
```
ComposeBox [nfc]: Add autoFocus props.
```

`useTopicAutocomplete` and `alwaysShowTopic` don't quite fit the description in the commit message.

Also, do you have a specific use for them in mind? It looks like we're not passing either one of them, in `ComposeBox`'s only callsite.

One thing I'd caution about is that there are cases where it'll be inappropriate to show anything about a topic: when editing a PM, and when composing a new PM to send.

Instead of these two props, I wonder if there could be just one prop that says whether the compose box should be aimed at working on a PM message or a stream message.

Maybe most of `getCanSelectTopic` (all but the `isFocused` part, I think) can be pulled out of `ComposeBox` to get the value for that prop? That seems like a refactoring commit that might go in the right direction. (Separately, though, I don't like the way we use `props.narrow` to decide whether the compose box should be aware of topics—eventually, I'd like to stop refusing to show the compose box in narrows like the all-messages narrow, from which reasonable users will want to edit a message without having to find it in a different narrow first. That'll mean inspecting the actual message the user wants to edit, in some appropriate place, to see if it's a stream message or a PM.)

Reply:
> `useTopicAutocomplete` and `alwaysShowTopic` don't quite fit the description in the commit message.

Ugh, these accidentally got mixed in when I was rebasing, I used them later in the new topic UI work. They're gone in my new revision.

I don't think that the refactor you describe would work for the new topic UI, although it seems like a reasonable idea in general.

The idea behind these was:

`useTopicAutocomplete`: When you're making a new topic, we don't want to autocomplete old topic names.
`alwaysShowTopic`: This makes it show the "Topic" input box even when the composebox isn't focused, which is desirable for the new topic UI, but not for general use. What we probably actually want is a `showTopicBox: 'always' | 'focus' | 'never';` with `focus` as the default, or something like that.

Reply:
Got it, makes sense.

> I wonder if there could be just one prop that says whether the compose box should be aimed at working on a PM message or a stream message.

I think exploring a prop like the one I've mentioned could be work for a different PR. I'm interested in looking into that at some point, if it sounds OK to you; I don't think there would be a lot of obstacles to adding it, but you're right that it wouldn't get you any farther toward making the new-topic UI.

For the new-topic UI, instead of `useTopicAutocomplete` and `alwaysShowTopic`, what do you think about a single prop like `isRequestingNewTopic` (or some better name)? If that sounds OK to you, I think it could be a good anchor for this part of `ComposeBox`'s interface, to help keep it steady until `ComposeBox` (eventually) has better support for PMs. When that time comes, we'll naturally be curious about adding the analogous "new PM" UI that the web app has alongside "new topic"—

![image](https://user-images.githubusercontent.com/22248748/121273580-8e1f5600-c896-11eb-839b-844e20578ca0.png)

—and I could imagine it being nice to add a simple `isRequestingNewPm` prop (or perhaps coalescing the two into `isRequestingNewDestinationNarrow` if that turns out to be worth considering).

Reply:
My reason for doing separate props was so that it would be clearer what they actually do, but I see the argument for bundling them as well.

Perhaps we should have a `mode: 'stream' | 'topic' | 'pm' | 'newTopic' | 'newPm';` or something like that? We wouldn't want to let someone set both `isRequestingNewPm` and `isRequestingNewTopic`.

Reply:
> My reason for doing separate props was so that it would be clearer what they actually do

Fair, although a good jsdoc can clarify what a prop does (and we're trying to be better about that) 🙂.

> Perhaps we should have a `mode: 'stream' | 'topic' | 'pm' | 'newTopic' | 'newPm';` or something like that? We wouldn't want to let someone set both `isRequestingNewPm` and `isRequestingNewTopic`.

Seems worth a try; I like that this means we don't have to think about the possibility of contradictory settings being set.